pkg/api/registry/registry.proto

```diff
@@ -22,61 +22,48 @@ message Destination {
     uint32 weight = 2;
 }
 
-message NetworkServiceManager {
-    string name = 1;
-    string url = 2;
-    google.protobuf.Timestamp expiration_time = 3;
-    string state = 4;
+message NetworkServiceLabels {
+    map<string, string> labels = 1;
 }
 
 message NetworkServiceEndpoint {
     string name = 1;
-    string payload = 2;
-    string network_service_name = 3;
-    string network_service_manager_name = 4;
-    map<string, string> labels = 5;
-    string state = 6;
-
+    repeated string network_service_name = 2;
+    map<string, NetworkServiceLabels> network_service_labels = 3;
     // Endpoint callback address, could be any of unix/tcp/callback schema ones.
-    string url = 7;
+    string url = 4;
+    // Network service manager URL for the endpoint.
+    string network_service_manger_url = 5;
+    google.protobuf.Timestamp expiration_time = 6;
 }
 
-message FindNetworkServiceRequest {
-    string network_service_name = 1;
+service NetworkServiceEndpointRegistry {
+    rpc Register (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
+    rpc Monitor (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
+    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpointList);
```

Comment:
We would stream here a Response of some sort, not a List.  The point of streaming is that we don't need a list :)

Reply:
> The point of streaming is that we don't need a list :)

In general, you are right, but lest consider the case when we need to send a million endpoints. So if we'll use steam of lists where list length is `length = (GRPC_MAX_MESSAGE_SIZE/sizeof(NetworkServiceEndpoint))` then it'll work in length time faster them single message stream.

Reply:
@edwarnicke make sense?

Reply:
Finally fixed.